hooks/gen_docs.py

```diff
@@ -1,10 +1,263 @@
+PATH_DOCS_COMPONENTS = PATH_DOCS_RESOURCES / "pipeline-components"
+PATH_DOCS_COMPONENTS_DEPENDENCIES = (
+    PATH_DOCS_COMPONENTS / "dependencies/pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS = (
+    PATH_DOCS_COMPONENTS / "dependencies/defaults_pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_KPOPS_STRUCTURE = PATH_DOCS_COMPONENTS / "dependencies/kpops_structure.yaml"
+KPOPS_COMPONENTS = tuple(_find_classes("kpops.components", PipelineComponent))
+KPOPS_COMPONENTS_INHERITANCE_REF: dict = {
+    component.get_component_type(): component.__base__.get_component_type()  # type: ignore[reportGeneralTypeIssues]
+    for component in KPOPS_COMPONENTS
+}
+KPOPS_COMPONENTS_FIELDS = {
+    # `set` to make it pickleable
+    component.get_component_type(): set(component.__fields__.keys())
+    for component in KPOPS_COMPONENTS
+}
+SECTIONS_ORDER = [
+    "type",
+    "name",
+    "namespace",
+    "app",
+    "from_",
+    "to",
+    "prefix",
+    "repo_config",
+    "version",
+    "resetter_values",
+    "offset_topic",
+]
+
+#####################
+# EXAMPLES          #
+#####################
+
+
+def filter_sections(
+    component_name: str, sections: list[str], include_inherited: bool = False
+) -> list[str]:
+    """Find all sections that are specific to a component from a list
+
+    :param component_name: Component name
+    :param sections: Available section files, names with extension, no path
+    :param include_inherited: Whether to include sections that can be defined in
+        a parent component defaults
+    :returns: The applicable sections
+    """
+    component_sections: list[str] = []
+    for target_section in KPOPS_COMPONENTS_FIELDS[component_name]:
+        component = component_name
+        section = target_section + "-" + component + ".yaml"
+        if section in sections:
+            component_sections.append(section)
+        elif (
+            not include_inherited
+            and KPOPS_COMPONENTS_INHERITANCE_REF[component] == "pipeline-component"
+        ):
+            section = target_section + ".yaml"
+            if section in sections:
+                component_sections.append(section)
+        elif include_inherited:
+            while component != "pipeline-component":
+                component = KPOPS_COMPONENTS_INHERITANCE_REF[component]
+                section = target_section + "-" + component + ".yaml"
+                if section in sections:
+                    component_sections.append(section)
+                    break
+                elif component == "pipeline-component":
+                    section = target_section + ".yaml"
+                    if section in sections:
+                        component_sections.append(section)
+                        break
+    component_sections.sort(key=component_section_position_in_definition)
+    return component_sections
+
+
+def concatenate_text_files(*file_paths: Path, target: Path) -> None:
```

Comment:
nice use of the star operator for unpacking! 👍🏻 
make `target` a kw-only argument, it's slightly nicer :)
```suggestion
def concatenate_text_files(*file_paths: Path, *, target: Path) -> None:
```

Reply:
also consider renaming `file_paths` to `sources` (or `source_paths` and `target_path` if you prefer the longer names)

Reply:
> nice use of the star operator for unpacking! 👍🏻 make `target` a kw-only argument, it's slightly safer :)

Thanks! It isn't allowed, the `*` arg makes it impossible to use `target` different from a keyword.
![image](https://github.com/bakdata/kpops/assets/44573859/035aa639-c9d1-4b2f-8267-6541203b1542)

Reply:
> 

done! reads much better